woocommerce/wc-admin
# Update payment method suggestion endpoint

This updates the payment method suggestions logic to use the new `payment-method/suggestions.json` endpoint that follows the same data format as our other specs. This allows the country and plugin validation to be moved to the specs, instead of being handled on the client.

I did have to update the endpoint a little bit, there is a separate PR for those changes.

### Screenshots

### Detailed test instructions:

- Load this branch and also the `update/payment-recommendations-endpoint` branch in woocommerce.com. 
- Make sure you have market place suggestions enabled.
- Go to **WooCommerce > Settings > Payments**
- Check if the **Recommended ways to get paid** shows up below the payment table with WcPay, Stripe, and Paypal (unless you have one installed).
- Click on **Get started** for one and make sure it installs correctly and redirects to the set up screen.
- Go back to the **Settings > Payments** screen, the card should now show up without the installed & activated plugin.
- Set your WooCommerce store to a WC Pay unsupported country (with WC Pay disabled ) you could use Brazil.
- Go back to **Settings > Payments**
- Make sure WC Pay doesn't show up in the list
- Set your WooCommerce store back to a supported WC Pay country
- Change the return value [here](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/WcPayPromotion/Init.php#L122) to `true`
- Reload the **Settings > Payments** screen, make sure that WooCommerce Payments shows in the payment method list with an **Install** button and the card does not display WC Pay.
- Open your console and make sure the `wcadmin_settings_payments_recommendations_pageview` track fired with props for each gateway`_displayed`, if you don't see the tracks run `localStorage.setItem( 'debug', 'wc-admin:*' );` in your console and refresh the page.
- Click on the 3 dots of the card and select **Hide this**
- The card should now be hidden, refresh the page and make sure this is persisted
- Look in your console again, the `wcadmin_settings_payments_recommendations_pageview` track should still be fired, but just with the woocommerce_payments prop for the experiment.
